auacm/app/modules/problem_manager/problem_timer.py

```diff
@@ -0,0 +1,134 @@
+import math
+import os
+import shutil
+import sys
+
+from app import config
+from app import database
+from app.modules.problem_manager import models
+from app.modules.submission_manager import judge
+from app.modules.submission_manager.models import MockSubmission
+from app.modules.submission_manager.judge_test import MockUploadFile
+
+
+PROBLEMS_DIR = os.path.join(config.DATA_FOLDER, 'problems')
+
+
+class Timer:
+    """Timer times all of the judge solutions for a problem."""
+
+    def __init__(self, problem):
+        self.problem = problem
+        self.path = os.path.join(PROBLEMS_DIR, str(problem.pid), 'test')
+        self.submission_path = os.path.join(config.TEMP_DIRECTORY, 'submits')
+
+    def _create_temp_directory(self):
+        """Attempts to create a temp directory.
+
+        In the event that we can't create the temp directory, an OSError will
+        be raised. In that case, you probably want to specify a different
+        directory to create the temp files in in config.py.
+        """
+        os.makedirs(os.path.join(self.submission_path, str(self.problem.pid)))
+
+    def _remove_temp_files(self):
+        """Removes the temporary files.
+
+        In the event that we can't remove the file tree that was created, an
+        OSError will be raised. In that case, you should probably choose a new
+        temp directory as well as delete the old temp directory yourself.
+        """
+        shutil.rmtree(self.submission_path)
+
+    def _run(self):
+        """Times all of the problems in the problem's test folder.
+
+        This will iterate through all of the files in the problem's test
+        folder. For each file, it creates a mock submission (so that it does
```

Comment:
Perhaps it should be called a solutions folder. The name seems to predate this, so it's probably out of scope.